client.go

```diff
@@ -151,7 +165,26 @@ func (c *Client) listen() {
 					}
 				}
 
-				app.forward(c.id, c.session, boot)
+				// TODO: Ugly. Find a better way to wrap m.data into {"args": m.data}.
+				body := map[string]map[string]string{}
+				argsMap := map[string]string{}
+				json.Unmarshal(boot, &argsMap)
+				body["args"] = argsMap
+
+				if getRequestHeaders, ok := c.broker.site.clientIDToHeaders[c.id]; ok {
+					headersMap := map[string]string{}
+					mergeHeaders(headersMap, *getRequestHeaders, *c.header)
```

Comment:
QQ: Is there something more behind creating the destination result here instead of allocating it in the merge function?

Just asking because because I find it unusual.

Reply:
This is outdated now, but agree that its unusual. No specific reason.